includes/admin/tools/export/export-actions.php

```diff
@@ -37,18 +38,20 @@ function give_process_batch_export_form() {
 
 	require_once GIVE_PLUGIN_DIR . 'includes/admin/tools/export/class-batch-export.php';
 
-	/**
+    $classname = give_clean($_REQUEST['class']);
```

Comment:
@ravinderk I'm afraid this isn't enough. It still means that the request can initialize **any** class. That's way too open. This should have a whitelist of classes that its able to construct.

Reply:
@JasonTheAdams good catch. I will implement it

Reply:
@JasonTheAdams We are using this feature in another add-on too:
- Gift Aid
- Recurring Donation
- Peer To Peer

We are not aware of classes upfront, so we can not whitelist all classes. What do you think about the validation class by `instanceof \Give_Batch_Export`?

Reply:
Yeah! That works! That's a good constraint. 👍

Reply:
fixed: https://github.com/impress-org/givewp/compare/427d62bb9327...42834be8b970

Reply:
fixed: https://github.com/impress-org/givewp/compare/42834be8b970...f44b25e891d8